python/external_config/scripts/external_runner.py

```diff
@@ -0,0 +1,221 @@
+    tank_folder = os.path.dirname(sgtk_file)
+    python_folder = os.path.dirname(tank_folder)
+    return python_folder
+
+def _import_py_file(python_path, name):
+    """
+    Helper which imports a python file and returns it.
+
+    :param str python_path: path where module is located
+    :param str name: name of py file (without extension)
+    :returns: python object
+    """
+    mfile, pathname, description = imp.find_module(name, [python_path])
+    try:
+        module = imp.load_module(name, mfile, pathname, description)
+    finally:
+        if mfile:
+            mfile.close()
+    return module
+
+
+def start_engine(
+    configuration_uri,
+    pipeline_config_id,
+    plugin_id,
+    engine_name,
+    entity_type,
+    entity_id,
+    bundle_cache_fallback_paths
+
+):
+    """
+    Bootstraps into an engine.
+
+    :param str configuration_uri: URI to bootstrap (for when pipeline config id is unknown).
+    :param int pipeline_config_id: Associated pipeline config id
+    :param str plugin_id: Plugin id to use for bootstrap
+    :param str engine_name: Engine name to launch
+    :param str entity_type: Entity type to launch
+    :param str entity_id: Entity id to launch
+    :param list bundle_cache_fallback_paths: List of bundle cache paths to include.
+    :raises: EngineStartupFailure on failure
+    """
+    try:
+        # log to file.
+        sgtk.LogManager().initialize_base_file_handler(engine_name)
+        logger = sgtk.LogManager.get_logger(LOGGER_NAME)
+        logger.debug("")
+        logger.debug("-=" * 60)
+        logger.debug("Preparing ToolkitManager for command cache bootstrap.")
+
+        # Setup the bootstrap manager.
+        manager = sgtk.bootstrap.ToolkitManager()
+        manager.plugin_id = plugin_id
+        manager.bundle_cache_fallback_paths = bundle_cache_fallback_paths
+
+        if pipeline_config_id:
+            # we have a pipeline config id to launch.
+            manager.do_shotgun_config_lookup = True
+            manager.pipeline_configuration = pipeline_config_id
+        else:
+            # launch a base uri. no need to look in sg for overrides.
+            manager.do_shotgun_config_lookup = False
+            manager.base_configuration = configuration_uri
+
+        logger.debug("Starting %s using entity %s %s", engine_name, entity_type, entity_id)
+        engine = manager.bootstrap_engine(
+            engine_name,
+            entity={"type": entity_type, "id": entity_id}
+        )
+        logger.debug("Engine %s started using entity %s %s", engine, entity_type, entity_id)
+
+    except Exception, e:
+        # We need to give the server a way to know that this failed due
+        # to an engine initialization issue. That will allow it to skip
+        # this config gracefully and log appropriately.
+        logger.exception("Could not bootstrap configuration")
+
+        # print to our logger so it's picked up by the main process
+        print traceback.format_exc()
```

Comment:
Seems strange to have a `print` here, why not having a handler to the logger which logs on stdout?

Reply:
i'll take a look. this is old code. Not sure a logger is needed. We only want to pass across error messages.